database/DatabaseExampleSwift/SignInViewController.swift

```diff
@@ -36,97 +36,150 @@ class SignInViewController: UIViewController, UITextFieldDelegate {
   }
 
   @IBAction func didTapEmailLogin(_ sender: AnyObject) {
-    if let email = self.emailField.text, let password = self.passwordField.text {
-      showSpinner({
-        FIRAuth.auth()?.signIn(withEmail: email, password: password) { (user, error) in
-          self.hideSpinner({
-            if let error = error {
-              self.showMessagePrompt(error.localizedDescription)
+  
+    guard let email = self.emailField.text, let password = self.passwordField.text else {
+          self.showMessagePrompt("email/password can't be empty")
+      return
+    }
+    
+    showSpinner{}
+    
+    // Sign user in
+    FIRAuth.auth()?.signIn(withEmail: email, password: password, completion: { (user, error) in
+      
+      self.hideSpinner{}
+      
+      guard let user = user, (error == nil) else {
+        self.showMessagePrompt(error!.localizedDescription)
+        return
+      }
+      
+      self.ref.child("users").child(user.uid).observeSingleEvent(of: .value, with: { (snapshot) in
+        
+        // Check if user already exists
+        guard !snapshot.exists() else {
+          self.performSegue(withIdentifier: "signIn", sender: nil)
+          return
+        }
+        
+        // Otherwise, create a user in the database
+        self.showTextInputPrompt(withMessage: "Username:") { (userPressedOK, username) in
+          
+          guard let username = username else {
+            self.showMessagePrompt("Username can't be empty")
+            return
+          }
+          
+          self.showSpinner{}
+          
+          let changeRequest = FIRAuth.auth()?.currentUser?.profileChangeRequest()
+          changeRequest?.displayName = username
+          
+          changeRequest?.commitChanges() { (error) in
+            
+            self.hideSpinner{}
+            guard error == nil else {
```

Comment:
If you're not binding any variables in this `guard` statement, why not just use an `if let`? That way you can avoid the (inconsequential, but ugly) force-unwrap later.